### ISSUE TITLE
Set timeout for parse request

### ISSUE THREAD

**Issue by smacker** [bblfsh/client-scala#78]
I'm running analyzis on many files some of them are big.
I see quite many
```
rpc error: code = DeadlineExceeded desc = context deadline exceeded
```
errors.

I would prefer to process slower but avoid these errors.

Timeout future similar to client-go could help here.

**Comment by bzz** [comment]
Different gRPC clients have different approaches for handling timing issues, as noted in [official doc](https://grpc.io/docs/guides/concepts.html#deadlinestimeouts).

Java client, that Scala uses underneath, has only a concept of `Deadlines` and does not have Timeouts (the difference is well explained in https://github.com/grpc/grpc-java/issues/1495#issuecomment-190273517).

Current impl of the scala-client use the defaults from [io/grpc/CallOptions.java#L44](https://github.com/grpc/grpc-java/blob/v1.17.0/core/src/main/java/) which AFAIK do not set any deadline for each request.

So the simplest thing we could do here is to expose such `Deadlines` (a fixed point in time) as an optional argument of `BblfshClient` constructor, along with host, port, etc. which does not seem to be that useful. Best possible effort would be to expose the Deadline at `Parse()` API call, so it can be set per-call, rather then per-client.  

Let me know if that would help.

> I would prefer to process slower but avoid these errors.

From the analysis of the batch processing logs [discussed elsewhere](https://github.com/bblfsh/javascript-driver/pull/57), it seems that the Deadline solution might not help as a solution, as the problem might be in `bblfshd` server when under a heavy load.

But please, let me know if there any particular files that reproduce the deadline exception.

**Comment by smacker** [comment]
`Deadline` in `Parse()` sounds good. Let me come back to it a little bit later.
I'll try to come up with some test case that we can use to reproduce the issue and validate the fix.

**Comment by bzz** [comment]
All the information above is related to grpc timeout/deadline which will workin only after a migration to V2 protocol that is currently "on hold"  in #76 

For v1 that scala client does support right now we expose a [ParseRequest.timeout](https://github.com/bblfsh/sdk/blob/v1/protocol/generated.proto#L55) on Protobuf level so under this issue I'm going to make sure it's configurable by the client.

This should reach the desired effect for current state of affairs.

**Comment by smacker** [comment]
Test case:
```
$ git clone https://github.com/smacker/client-scala
$ cd client-scala
$ git checkout timeout_error

smacker in ~/Work/client-scala on timeout_error [?]
$ ./sbt "test-only *BblfshClientTimeoutTest"
[info] BblfshClientTimeoutTest:
[info] - Parse timeout requests *** FAILED ***
[info]   Vector("rpc error: code = DeadlineExceeded desc = context deadline exceeded") was not empty (BblfshClientTimeoutTest.scala:21)
```
